src/main/java/org/opensrp/repository/postgres/IdentifierSourceRepositoryImpl.java

```diff
@@ -0,0 +1,165 @@
+		throw new NotImplementedException();
+	}
+
+	@Override
+	protected Long retrievePrimaryKey(IdentifierSource entity) {
+
+		IdentifierSourceExample identifierSourceExample = new IdentifierSourceExample();
+		IdentifierSourceExample.Criteria criteria = identifierSourceExample.createCriteria();
+		criteria.andIdentifierIsNotNull();
+
+		criteria.andIdEqualTo(entity.getId() != null ? entity.getId() : 0);
+		org.opensrp.domain.postgres.IdentifierSource identifierSource = customIdentifierSourceMapper
+				.selectOne(identifierSourceExample);
+		if (identifierSource == null) {
+			return null;
+		}
+		return identifierSource.getId();
+	}
+
+	@Override
+	protected Object getUniqueField(IdentifierSource identifierSource) {
+		if (identifierSource == null) {
+			return null;
+		}
+		return identifierSource.getIdentifier();
+	}
+
+	// Private Methods
+	protected List<IdentifierSource> convert(List<org.opensrp.domain.postgres.IdentifierSource> identifierSources) {
+		if (identifierSources == null || identifierSources.isEmpty()) {
+			return new ArrayList<>();
+		}
+
+		List<IdentifierSource> convertedIdentifierSources = new ArrayList<>();
+		for (org.opensrp.domain.postgres.IdentifierSource identifierSource : identifierSources) {
+			IdentifierSource convertedIdentifierSource = convert(identifierSource);
+			if (convertedIdentifierSource != null) {
+				convertedIdentifierSources.add(convertedIdentifierSource);
+			}
+		}
+
+		return convertedIdentifierSources;
+	}
+
+	private IdentifierSource convert(org.opensrp.domain.postgres.IdentifierSource identifierSource) {
+		if (identifierSource == null) {
+			return null;
+		}
+		IdentifierSource convertedIdentifierSource = new IdentifierSource();
+		convertedIdentifierSource.setId(identifierSource.getId());
+		convertedIdentifierSource.setIdentifier(identifierSource.getIdentifier());
+		convertedIdentifierSource.setDescription(identifierSource.getDescription());
+		convertedIdentifierSource.setIdentifierValidatorAlgorithm(IdentifierValidatorAlgorithm.get(identifierSource.getIdentifierValidatorAlgorithm()));
+		convertedIdentifierSource.setBaseCharacterSet(identifierSource.getBaseCharacterSet());
+		convertedIdentifierSource.setFirstIdentifierBase(identifierSource.getFirstIdentifierBase());
+		convertedIdentifierSource.setPrefix(identifierSource.getPrefix());
+		convertedIdentifierSource.setSuffix(identifierSource.getSuffix());
+		convertedIdentifierSource.setMinLength(identifierSource.getMinLength());
+		convertedIdentifierSource.setMaxLength(identifierSource.getMaxLength());
+		convertedIdentifierSource.setBlacklisted(identifierSource.getBlacklisted());
+
+		return convertedIdentifierSource;
+	}
+
+	private org.opensrp.domain.postgres.IdentifierSource convert(IdentifierSource identifierSource, Long primaryKey) {
+		if (identifierSource == null) {
+			return null;
+		}
+
+		org.opensrp.domain.postgres.IdentifierSource pgIdentifierSource = new org.opensrp.domain.postgres.IdentifierSource();
```

Comment:
can we use builder to make it cleaner? or may be not... as all other code is written in same way..

Reply:
This is auto generated class, so adding builder would always be overwritten when generating again. A solution would be to extend the generated class.